extension/encoding/googlecloudlogentryencodingextension/log_entry.go

```diff
@@ -2,27 +2,244 @@
 // SPDX-License-Identifier: Apache-2.0
 
 package googlecloudlogentryencodingextension // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/encoding/googlecloudlogentryencodingextension"
-
 import (
 	"encoding/hex"
-	stdjson "encoding/json"
 	"errors"
 	"fmt"
+	"strconv"
 	"strings"
-	"sync"
 	"time"
 
-	"cloud.google.com/go/logging/apiv2/loggingpb"
+	gojson "github.com/goccy/go-json"
 	"github.com/iancoleman/strcase"
-	jsoniter "github.com/json-iterator/go"
 	"go.opentelemetry.io/collector/pdata/pcommon"
 	"go.opentelemetry.io/collector/pdata/plog"
-	"google.golang.org/genproto/googleapis/api/monitoredres"
-	"google.golang.org/protobuf/encoding/protojson"
-	"google.golang.org/protobuf/reflect/protoreflect"
 )
 
-var json = jsoniter.ConfigCompatibleWithStandardLibrary
+// See: https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry
+type logEntry struct {
```

Comment:
Is there a reason to not use the upstream [definitions](https://github.com/googleapis/google-cloud-go/blob/main/logging/apiv2/loggingpb/log_entry.pb.go)?

Reply:
I think I understand what you mean now. You want us to look for it, instead of defining the JSON struct?

There are two big reasons to not go there:
1. The issue this is related to: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/41087. We want to use semantic conventions, so we need to define these fields so we can map them correctly.
2. Performance is very poor, I posted the results on the issue description

Reply:
I understand what you mean now.

There are fields missing from the definitions: app hub and error groups, for example